services/rfq/api/rest/server.go

```diff
@@ -278,18 +278,22 @@ func (r *QuoterAPIServer) checkRole(c *gin.Context, destChainID uint32) (address
 
 	if hasRole == nil || hasRole.IsExpired() {
 		has, roleErr := bridge.HasRole(ops, relayerRole, addressRecovered)
-		if roleErr == nil {
-			r.roleCache[destChainID].Set(addressRecovered.Hex(), has, cacheInterval)
-		}
-
 		if roleErr != nil {
 			err = fmt.Errorf("unable to check relayer role on-chain")
 			return addressRecovered, err
-		} else if !has {
-			err = fmt.Errorf("q.Relayer not an on-chain relayer")
+		}
+
+		r.roleCache[destChainID].Set(addressRecovered.Hex(), has, cacheInterval)
+
+		if !has {
+			err = fmt.Errorf("relayer not an on-chain relayer")
 			return addressRecovered, err
 		}
+	} else if !hasRole.Value() {
+		err = fmt.Errorf("relayer not an on-chain relayer")
+		return addressRecovered, err
```

Comment:
@aureliusbtc I would separate the "check/populate the cache" and "check the `has` value" steps here. This would remove the duplication of the `relayer not an on-chain relayer` error, and make it easier to maintain later (e.g. add tracing)

Reply:
Also, imo `hasRole` and `has` could be better named to reflect that the first one is actually a cache